incubator/hnc/pkg/validators/hierarchy.go

```diff
@@ -127,6 +132,20 @@ func (v *Hierarchy) checkForest(hc *api.HierarchyConfiguration) ([]authzReq, adm
 	curParent := ns.Parent()
 	newParent := v.Forest.Get(hc.Spec.Parent)
 
+	// Check if requiredChild names in HierarchyConfiguration.Spec.RequiredChildren obey
+	// kubernetes namespace regex format.
+	invalidRCs := []string{}
+	validRCRex := regexp.MustCompile(validRCRexFmt)
+	for _, rc := range hc.Spec.RequiredChildren {
+		if !validRCRex.MatchString(rc) || len(rc) > validRCMaxLength {
+			invalidRCs = append(invalidRCs, rc)
+		}
+	}
+
+	if len(invalidRCs) > 0 {
+		return nil, deny(metav1.StatusReasonBadRequest, fmt.Sprintf("Invalid requiredChild name: %v", strings.Join(invalidRCs, " ")))
```

Comment:
Better: maybe something like "The following required children are not valid namespace names: "

Reply:
I see. WIll revised it soon. :)